Set up skeleton repo

jonathanpallant opened the pull request:
Has an example that compiles and runs using semihosting.

jonathanpallant committed:
Set up skeleton repo

Has an example that compiles and runs using semihosting.

jonathanpallant committed:
Add clippy and rustfmt to CI.

jonathanpallant committed:
Add Ferrocene build

jonathanpallant committed:
Use token

Hoverbear commented on .github/workflows/build.yml:
```diff
@@ -0,0 +1,109 @@
+    strategy:
+      matrix:
+        rust: [stable, 1.76]
+        TARGET:
```
The choice of capitalizing one and not the other is a bit odd, but not a problem.

jonathanpallant committed:
Consistent capitalisation

jonathanpallant committed:
Now the FPU is conditionally enabled.

The conditions are that you pick a hard-float target, or you use the "eabi-fpu" feature I added. You must use this feature when specifying "target-cpu=cortex-r5" becuase that enables the use of FPU instructions.

jonathanpallant committed:
OK, rustfmt and I will just agree to disagree

Hoverbear commented on cortex-r-semihosting/src/hio.rs:
```diff
@@ -0,0 +1,69 @@
+//! Host I/O
+
+// Fixing this lint requires a breaking change that does not add much value
+#![allow(clippy::result_unit_err)]
```
This is a newly released crate, can we make that breaking change so we can drop this?

Hoverbear commented on cortex-r-semihosting/src/macros.rs:
```diff
@@ -0,0 +1,120 @@
+/// Macro that prints and returns the value of a given expression for quick and
+/// dirty debugging.
+///
+/// Works exactly like `dbg!` in the standard library, replacing `eprintln!`
```
Colliding with a `std` name is a bit odd given that `heprin` etc go out of their way to avoid it, is there a specific reason for this choice?

jonathanpallant replied:
I'm not minded to change it, so that it matches what cortex-m-semihosting does (which this will hopefully merge with one day)

jonathanpallant committed:
Remove unit errors from cortex-r-semihosting
